### PR TITLE
Added nullable annotations to Avalonia.Visuals.

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

Adds nullable annotations to Avalonia.Visuals. Done with help from @Gillibald who annotated the text-related stuff.  

## Breaking changes

The default `GlyphRun` constructor was removed, but given that using this constructor would have just given you a non-usable `GlyphRun` I'm not sure how breaking it is in reality. If we want to port this to the stable branch we could put it back in.

### CODE REVIEW

**Comment by Takoooooo** [comment]
@grokys there are lots of tests failing and also a merge conflict

File: src/Avalonia.Visuals/Media/DrawingContext.cs [hunk 1]
```diff
@@ -317,7 +323,14 @@ public PushedState PushClip(Rect clip)
...
         public PushedState PushGeometryClip(Geometry clip)
         {
-            Contract.Requires<ArgumentNullException>(clip != null);
+            _ = clip ?? throw new ArgumentNullException(nameof(clip));
+
+            // HACK: This check was added when nullable annotations pointed out that we're potentially
+            // pushing a null value for the clip here. Ideally we'd return an empty PushedState here but
+            // I don't want to make that change as part of adding nullable annotations.
+            if (clip.PlatformImpl is null)
+                throw new InvalidOperationException("Cannot push empty geometry clip.");
```
Comment (grokys) [inline src/Avalonia.Visuals/Media/DrawingContext.cs:332]: This needs fixing properly, but not here, and with unit tests.

File: src/Avalonia.Visuals/Media/PathMarkupParser.cs [hunk 2]
```diff
@@ -98,6 +99,8 @@ private static Point MirrorControlPoint(Point controlPoint, Point center)
         /// <param name="pathData">The path data.</param>
         public void Parse(string pathData)
         {
+            ThrowIfDisposed();
```
Comment (grokys) [inline src/Avalonia.Visuals/Media/PathMarkupParser.cs:102]: Hopefully adding these checks won't slow down path parsing too much?

Comment (maxkatz6) [reply src/Avalonia.Visuals/Media/PathMarkupParser.cs:102]: Isn't Path parsed in compile time now?

File: src/Android/Avalonia.AndroidTestApplication/Resources/Resource.Designer.cs [hunk 3]
```diff
@@ -14,7 +14,7 @@ namespace Avalonia.AndroidTestApplication
 {
 	
 	
-	[global::System.CodeDom.Compiler.GeneratedCodeAttribute("Xamarin.Android.Build.Tasks", "1.0.0.0")]
+	[global::System.CodeDom.Compiler.GeneratedCodeAttribute("Xamarin.Android.Build.Tasks", "12.1.99.62")]
```
Comment (maxkatz6) [inline src/Android/Avalonia.AndroidTestApplication/Resources/Resource.Designer.cs:17]: nit: unrelated

File: src/Avalonia.Build.Tasks/Avalonia.Build.Tasks.csproj [hunk 4]
```diff
@@ -6,7 +6,7 @@
         <BuildOutputTargetFolder>tools</BuildOutputTargetFolder>
         <DefineConstants>$(DefineConstants);BUILDTASK;XAMLX_CECIL_INTERNAL;XAMLX_INTERNAL</DefineConstants>
         <CopyLocalLockFileAssemblies Condition="$(TargetFramework) == 'netstandard2.0'">true</CopyLocalLockFileAssemblies>
-        <NoWarn>NU1605</NoWarn>
+        <NoWarn>NU1605;CS8632</NoWarn>
```
Comment (maxkatz6) [inline src/Avalonia.Build.Tasks/Avalonia.Build.Tasks.csproj:9]: Why do we need it?

Comment (grokys) [reply src/Avalonia.Build.Tasks/Avalonia.Build.Tasks.csproj:9]: Ah yeah, it's because Avalonia.Build.Tasks doesn't enable nullability but it does include files via links which have nullable annotations.

File: src/Android/Avalonia.AndroidTestApplication/Resources/Resource.Designer.cs (hunk 3 above)
Comment (grokys) [reply src/Android/Avalonia.AndroidTestApplication/Resources/Resource.Designer.cs:17]: Grr visual studio keeps trying to sneak these in. Looks like this time it succeeded. Will revert.

**Comment by grokys** [comment]
Fixed the merge conflict, need to work out how I broke render tests.

File: src/Avalonia.Visuals/Platform/IFontManagerImpl.cs [hunk 5]
```diff
@@ -31,7 +31,7 @@ public interface IFontManagerImpl
         /// </returns>
         bool TryMatchCharacter(int codepoint, FontStyle fontStyle,
             FontWeight fontWeight,
-            FontFamily fontFamily, CultureInfo culture, out Typeface typeface);
+            FontFamily? fontFamily, CultureInfo? culture, out Typeface typeface);
```
Comment (maxkatz6) [inline src/Avalonia.Visuals/Platform/IFontManagerImpl.cs:34]: Can't typeface be null when this method returns false?

Comment (Gillibald) [reply src/Avalonia.Visuals/Platform/IFontManagerImpl.cs:34]: Typeface is a struct and will have its default value.

**Review by maxkatz6 — APPROVED** [review]

File: src/Android/Avalonia.AndroidTestApplication/Resources/Resource.Designer.cs (hunk 3 above)
Comment (maxkatz6) [reply src/Android/Avalonia.AndroidTestApplication/Resources/Resource.Designer.cs:17]: It's fine, otherwise VS will never shut up
